File: sticker.ts
// sticker.ts
// Sticker Maker на TypeScript (использует @imgly/background-removal)

import * as fs from 'fs';
import { removeBackground } from '@imgly/background-removal';
import Jimp from 'jimp';

async function main(): Promise<void> {
    const args = process.argv.slice(2);
    let input: string | null = null;
    let output: string | null = null;
    let color: string | null = null;
    let threshold: number = 30;
    let verbose: boolean = false;

    for (let i = 0; i < args.length; i++) {
        switch (args[i]) {
            case '-o':
            case '--output':
                output = args[++i];
                break;
            case '-c':
            case '--color':
                color = args[++i];
                break;
            case '-t':
            case '--threshold':
                threshold = parseInt(args[++i]);
                break;
            case '-v':
            case '--verbose':
                verbose = true;
                break;
            case '-h':
            case '--help':
                console.log(`Использование: ts-node sticker.ts [опции] <входной_файл>
  -o, --output <файл>   Выходной файл (PNG)
  -c, --color <HEX>     Цвет фона (не используется)
  -t, --threshold <N>   Порог (не используется)
  -v, --verbose         Подробный вывод
  -h, --help            Справка`);
                process.exit(0);
            default:
                input = args[i];
        }
    }

    if (!input) {
        console.error('Ошибка: не указан входной файл.');
        process.exit(1);
    }

    if (!fs.existsSync(input)) {
        console.error(`Ошибка: файл ${input} не найден`);
        process.exit(1);
    }

    output = output || input.replace(/\.[^.]+$/, '') + '_sticker.png';

    try {
        const image = await Jimp.read(input);
        const base64 = await image.getBase64Async(Jimp.MIME_PNG);
        const resultBlob = await removeBackground(base64);
        const buffer = Buffer.from(await resultBlob.arrayBuffer());
        fs.writeFileSync(output, buffer);
        if (verbose) {
            console.log(`Фон удалён. Результат сохранён в ${output}`);
        }
    } catch (err: any) {
        console.error(`Ошибка обработки: ${err.message}`);
        process.exit(1);
    }
}

main();

File: sticker.py
# sticker.py
# Sticker Maker на Python (использует rembg)

import sys
import os
import argparse
from PIL import Image
from rembg import remove

def main():
    parser = argparse.ArgumentParser(description="Sticker Maker (Background Removal) - Python")
    parser.add_argument("input", help="Входной файл")
    parser.add_argument("-o", "--output", help="Выходной файл (PNG)")
    parser.add_argument("-c", "--color", help="Цвет фона (HEX) - не используется в rembg, оставлен для совместимости")
    parser.add_argument("-t", "--threshold", type=int, default=30, help="Порог (не используется)")
    parser.add_argument("-v", "--verbose", action="store_true", help="Подробный вывод")
    args = parser.parse_args()

    if not os.path.exists(args.input):
        print(f"Ошибка: файл {args.input} не найден")
        sys.exit(1)

    output = args.output or os.path.splitext(args.input)[0] + "_sticker.png"

    try:
        with open(args.input, 'rb') as i:
            input_data = i.read()
            output_data = remove(input_data)
            with open(output, 'wb') as o:
                o.write(output_data)
        if args.verbose:
            print(f"Фон удалён. Результат сохранён в {output}")
    except Exception as e:
        print(f"Ошибка обработки: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()
